ffi-yajl doesn't compile on MRI 2.4.0-preview1

```
[ (ruby-2.4.0-preview1) cwimmer@athos ~]$ rvm gemset use test1 --create
Using ruby-2.4.0-preview1 with gemset test1
ruby-2.4.0-preview1 - #gemset created /Users/cwimmer/.rvm/gems/ruby-2.4.0-preview1@test1
ruby-2.4.0-preview1 - #generating test1 wrappers..........
[ (ruby-2.4.0-preview1@test1) cwimmer@athos ~]$ gem install ffi-yajl -v '2.2.3'
Fetching: libyajl2-1.2.0.gem (100%)
Building native extensions.  This could take a while...
Successfully installed libyajl2-1.2.0
Fetching: ffi-yajl-2.2.3.gem (100%)
Building native extensions.  This could take a while...
ERROR:  Error installing ffi-yajl:
	ERROR: Failed to build gem native extension.

    current directory: /Users/cwimmer/.rvm/gems/ruby-2.4.0-preview1@test1/gems/ffi-yajl-2.2.3/ext/ffi_yajl/ext/encoder
/Users/cwimmer/.rvm/rubies/ruby-2.4.0-preview1/bin/ruby -r ./siteconf20160707-17231-p41c47.rb extconf.rb
 -I/Users/cwimmer/.rvm/gems/ruby-2.4.0-preview1@test1/gems/libyajl2-1.2.0/lib/libyajl2/vendored-libyajl2/include $(cflags)  -fno-common -pipe
 -L/Users/cwimmer/.rvm/gems/ruby-2.4.0-preview1@test1/gems/libyajl2-1.2.0/lib/libyajl2/vendored-libyajl2/lib -L. -fstack-protector -L/usr/local/lib -L/usr/local/opt/libyaml/lib -L/usr/local/opt/readline/lib -L/usr/local/opt/libksba/lib -L/usr/local/opt/openssl/lib
checking for yajl/yajl_tree.h... yes
creating Makefile

To see why this extension failed to compile, please check the mkmf.log which can be found here:

  /Users/cwimmer/.rvm/gems/ruby-2.4.0-preview1@test1/extensions/x86_64-darwin-12/2.4.0/ffi-yajl-2.2.3/mkmf.log

current directory: /Users/cwimmer/.rvm/gems/ruby-2.4.0-preview1@test1/gems/ffi-yajl-2.2.3/ext/ffi_yajl/ext/encoder
make "DESTDIR=" clean

current directory: /Users/cwimmer/.rvm/gems/ruby-2.4.0-preview1@test1/gems/ffi-yajl-2.2.3/ext/ffi_yajl/ext/encoder
make "DESTDIR="
compiling encoder.c
encoder.c:94:13: warning: implicit conversion loses integer precision: 'long' to 'int' [-Wshorten-64-to-32]
  int len = RSTRING_LEN(str);
            ^~~~~~~~~~~~~~~~
/Users/cwimmer/.rvm/rubies/ruby-2.4.0-preview1/include/ruby-2.4.0/ruby/ruby.h:965:6: note: expanded from macro 'RSTRING_LEN'
     RSTRING_EMBED_LEN(str) : \
     ^
/Users/cwimmer/.rvm/rubies/ruby-2.4.0-preview1/include/ruby-2.4.0/ruby/ruby.h:961:6: note: expanded from macro 'RSTRING_EMBED_LEN'
     (long)((RBASIC(str)->flags >> RSTRING_EMBED_LEN_SHIFT) & \
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
encoder.c:94:13: warning: implicit conversion loses integer precision: 'long' to 'int' [-Wshorten-64-to-32]
  int len = RSTRING_LEN(str);
            ^~~~~~~~~~~~~~~~
/Users/cwimmer/.rvm/rubies/ruby-2.4.0-preview1/include/ruby-2.4.0/ruby/ruby.h:966:28: note: expanded from macro 'RSTRING_LEN'
     RSTRING(str)->as.heap.len)
                           ^
encoder.c:110:13: warning: implicit conversion loses integer precision: 'long' to 'int' [-Wshorten-64-to-32]
  int len = RSTRING_LEN(str);
            ^~~~~~~~~~~~~~~~
/Users/cwimmer/.rvm/rubies/ruby-2.4.0-preview1/include/ruby-2.4.0/ruby/ruby.h:965:6: note: expanded from macro 'RSTRING_LEN'
     RSTRING_EMBED_LEN(str) : \
     ^
/Users/cwimmer/.rvm/rubies/ruby-2.4.0-preview1/include/ruby-2.4.0/ruby/ruby.h:961:6: note: expanded from macro 'RSTRING_EMBED_LEN'
     (long)((RBASIC(str)->flags >> RSTRING_EMBED_LEN_SHIFT) & \
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
encoder.c:110:13: warning: implicit conversion loses integer precision: 'long' to 'int' [-Wshorten-64-to-32]
  int len = RSTRING_LEN(str);
            ^~~~~~~~~~~~~~~~
/Users/cwimmer/.rvm/rubies/ruby-2.4.0-preview1/include/ruby-2.4.0/ruby/ruby.h:966:28: note: expanded from macro 'RSTRING_LEN'
     RSTRING(str)->as.heap.len)
                           ^
encoder.c:262:30: warning: passing 'const char [1]' to parameter of type 'char *' discards qualifiers [-Wincompatible-pointer-types]
    gen_cstring(rb_yajl_gen, "", sizeof("")-1);
                             ^~
encoder.c:79:44: note: passing argument to parameter 'cptr' here
VALUE gen_cstring(VALUE rb_yajl_gen, char *cptr, int len) {
                                           ^
encoder.c:272:30: warning: passing 'const char [5]' to parameter of type 'char *' discards qualifiers [-Wincompatible-pointer-types]
    gen_cstring(rb_yajl_gen, "true", sizeof("true")-1);
                             ^~~~~~
encoder.c:79:44: note: passing argument to parameter 'cptr' here
VALUE gen_cstring(VALUE rb_yajl_gen, char *cptr, int len) {
                                           ^
encoder.c:282:30: warning: passing 'const char [6]' to parameter of type 'char *' discards qualifiers [-Wincompatible-pointer-types]
    gen_cstring(rb_yajl_gen, "false", sizeof("false")-1);
                             ^~~~~~~
encoder.c:79:44: note: passing argument to parameter 'cptr' here
VALUE gen_cstring(VALUE rb_yajl_gen, char *cptr, int len) {
                                           ^
encoder.c:380:20: error: use of undeclared identifier 'rb_cFixnum'
  rb_define_method(rb_cFixnum, "ffi_yajl", rb_cFixnum_ffi_yajl, 2);
                   ^
encoder.c:381:20: error: use of undeclared identifier 'rb_cBignum'
  rb_define_method(rb_cBignum, "ffi_yajl", rb_cBignum_ffi_yajl, 2);
                   ^
7 warnings and 2 errors generated.
make: *** [encoder.o] Error 1

make failed, exit code 2

Gem files will remain installed in /Users/cwimmer/.rvm/gems/ruby-2.4.0-preview1@test1/gems/ffi-yajl-2.2.3 for inspection.
Results logged to /Users/cwimmer/.rvm/gems/ruby-2.4.0-preview1@test1/extensions/x86_64-darwin-12/2.4.0/ffi-yajl-2.2.3/gem_make.out
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -26,8 +26,6 @@ matrix:
       jdk: oraclejdk8
     - rvm: jruby-head
       jdk: openjdk7
-  allow_failures:
-    - rvm: ruby-head
 
 branches:
   only:

--- a/ext/ffi_yajl/ext/encoder/encoder.c
+++ b/ext/ffi_yajl/ext/encoder/encoder.c
@@ -377,8 +377,12 @@ void Init_encoder() {
   rb_define_method(rb_cNilClass, "ffi_yajl", rb_cNilClass_ffi_yajl, 2);
   rb_define_method(rb_cTrueClass, "ffi_yajl", rb_cTrueClass_ffi_yajl, 2);
   rb_define_method(rb_cFalseClass, "ffi_yajl", rb_cFalseClass_ffi_yajl, 2);
+#ifdef rb_cFixnum /* ruby < 2.4 */
   rb_define_method(rb_cFixnum, "ffi_yajl", rb_cFixnum_ffi_yajl, 2);
   rb_define_method(rb_cBignum, "ffi_yajl", rb_cBignum_ffi_yajl, 2);
+#else
+  rb_define_method(rb_cInteger, "ffi_yajl", rb_cFixnum_ffi_yajl, 2);
+#endif
   rb_define_method(rb_cFloat, "ffi_yajl", rb_cFloat_ffi_yajl, 2);
   rb_define_method(rb_cString, "ffi_yajl", rb_cString_ffi_yajl, 2);
   rb_define_method(rb_cSymbol, "ffi_yajl", rb_cSymbol_ffi_yajl, 2);

